src/fluree/db/flake/flake_db.cljc

```diff
@@ -292,6 +293,131 @@
           (merge-flakes t-new all-flakes)
           (assoc :commit commit-metadata)))))
 
+(defn- component->user-value
+  "Convert an internal pattern component to user-readable format"
+  [db component context]
+  (cond
+    (where/unmatched-var? component)
+    (str (where/get-variable component))
+
+    (where/matched-iri? component)
+    (let [iri (where/get-iri component)]
```

Comment:
I think we should use the actual `json-ld/compact` function here instead of this ad-hoc one, which will not handle nested aliases and possibly other edge cases.